zkevm-circuits/src/bytecode_circuit/circuit.rs

```diff
@@ -467,11 +509,56 @@ impl<F: Field> BytecodeCircuitConfig<F> {
                     self.set_padding_row(
                         &mut region,
                         &push_data_left_is_zero_chip,
+                        &index_length_diff_is_zero_chip,
                         empty_hash,
                         idx,
                         last_row_offset,
                     )?;
                 }
+
+                // Overwrite
+                let mut value_rlc = challenges.keccak_input().map(|_| F::zero());
+                for (offset, row) in overwrite.rows.iter().enumerate() {
+                    for (name, column, value) in [
+                        ("tag", self.bytecode_table.tag, row.tag),
+                        ("index", self.bytecode_table.index, row.index),
+                        ("is_code", self.bytecode_table.is_code, row.is_code),
+                        ("value", self.bytecode_table.value, row.value),
+                        ("length", self.length, F::from(overwrite.bytes.len() as u64)),
+                    ] {
+                        region.assign_advice(
+                            || format!("assign {} {}", name, offset),
+                            column,
+                            offset,
+                            || Value::known(value),
+                        )?;
+                    }
+
+                    if row.tag == F::one() {
```

Comment:
If we are overriding, shouldn't we also override the value_rlc or leave it alone alone as it was before?

Perhaps we could implement a system that allows to choose which columns we override?

Reply:
> If we are overriding, shouldn't we also override the value_rlc or leave it alone alone as it was before?

I wrote this overwrite code originally.  The reason why `value_rlc` is not included in the overwrite values is because it can't be precomputed as it depends on the challange, which is only known at phase 2.  So the lines below compute a correct value_rlc during assignment.

> Perhaps we could implement a system that allows to choose which columns we override?

This is the approach taken in the state circuit: https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/0e76d2b25e62c22f75b26a64d9d7ea78bb3fede0/zkevm-circuits/src/state_circuit.rs#L500
Nevertheless I'm not sure if it would be useful here. (I think it would make the negative tests more complex)

Reply:
Ok